src/NodeServices/PythonServices.cs

```diff
@@ -111,12 +111,52 @@ public sealed class PythonEngineManager
         public static PythonEngineManager Instance { get { return lazy.Value; } }
         #endregion
 
-        //TODO see DYN-6550 when hiding/replacing this obsolete field.
+        /// <summary>
+        /// A readonly collection of all the loaded Python engines
+        /// </summary>
+        public ReadOnlyCollection<PythonEngine> PythonEngines => new ReadOnlyCollection<PythonEngine>(AvailableEngines);
+
         /// <summary>
         /// An observable collection of all the loaded Python engines
         /// </summary>
-        [Obsolete("AvailableEngines field will be replaced in a future Dynamo release.")]
-        public ObservableCollection<PythonEngine> AvailableEngines;
+        internal ObservableCollection<PythonEngine> AvailableEngines;
+
+        public delegate void PythonEngineChangedHandler(PythonEngine engine);
+
+        private Action<PythonEngine> customizeEngine;
+
+        /// <summary>
+        /// Use this function to customize Python engine initialization.
+        /// This function will be called only once on all existing or future python engines (on PythonEngineAdded).
+        /// </summary>
+        public Action<PythonEngine> CustomizeEngine
```

Comment:
this seems kind of dangerous, I imagine you want integrations like Revit to use this to setup marshallers and hook up events?

But this would also be easily accessible to anyone since this class has a static instance method - it could easily break python integration.